### PR TITLE
Adds energy spear. [Reopening in New PR]

### PR DESCRIPTION
**Pull request by MisterLayne**
Adds a spear, except it's got energy. Ayy.

### CODE REVIEW

File: code/game/objects/items/weapons/melee/energy.dm [hunk 1]
```diff
@@ -58,9 +58,9 @@
 
 /obj/item/weapon/melee/energy/suicide_act(mob/user)
 	var/tempgender = "[user.gender == MALE ? "he's" : user.gender == FEMALE ? "she's" : "they are"]"
-	if (active)
-		viewers(user) << pick("<span class='danger'>\The [user] is slitting \his stomach open with \the [src]! It looks like [tempgender] trying to commit seppuku.</span>", \
-		                      "<span class='danger'>\The [user] is falling on \the [src]! It looks like [tempgender] trying to commit suicide.</span>")
+	if(active)
+		to_chat(viewers(user), pick("<span class='danger'>\The [user] is slitting \his stomach open with \the [src]! It looks like [tempgender] trying to commit seppuku.</span>", \
+		                      		"<span class='danger'>\The [user] is falling on \the [src]! It looks like [tempgender] trying to commit suicide.</span>"))
```
Comment (MisterLayne) [inline code/game/objects/items/weapons/melee/energy.dm]: Think the indentation mess-up is somewhere around here. No idea how to format this, though.

Comment (Anewbe) [reply code/game/objects/items/weapons/melee/energy.dm]: Replace that to_chat with visible_message, maybe?

File: code/game/objects/items/weapons/melee/energy.dm [hunk 2]
```diff
@@ -58,9 +58,9 @@
 
 /obj/item/weapon/melee/energy/suicide_act(mob/user)
 	var/tempgender = "[user.gender == MALE ? "he's" : user.gender == FEMALE ? "she's" : "they are"]"
-	if (active)
-		viewers(user) << pick("<span class='danger'>\The [user] is slitting \his stomach open with \the [src]! It looks like [tempgender] trying to commit seppuku.</span>", \
-		                      "<span class='danger'>\The [user] is falling on \the [src]! It looks like [tempgender] trying to commit suicide.</span>")
+	if(active)
+		user.visible_message("<span class='danger'>\The [user] is slitting \his stomach open with \the [src]! It looks like [tempgender] trying to commit seppuku.</span>",
+							"<span class='danger'>\The [user] is falling on \the [src]! It looks like [tempgender] trying to commit suicide.</span>")
```
Comment (MisterLayne) [inline code/game/objects/items/weapons/melee/energy.dm]: I did not even indent it that much, are you drunk github

File: code/game/objects/items/weapons/melee/energy.dm [hunk 3]
```diff
@@ -58,9 +58,9 @@
 
 /obj/item/weapon/melee/energy/suicide_act(mob/user)
 	var/tempgender = "[user.gender == MALE ? "he's" : user.gender == FEMALE ? "she's" : "they are"]"
-	if (active)
-		viewers(user) << pick("<span class='danger'>\The [user] is slitting \his stomach open with \the [src]! It looks like [tempgender] trying to commit seppuku.</span>", \
-		                      "<span class='danger'>\The [user] is falling on \the [src]! It looks like [tempgender] trying to commit suicide.</span>")
+	if(active)
+		user.visible_message(pick("<span class='danger'>\The [user] is slitting \his stomach open with \the [src]! It looks like [tempgender] trying to commit seppuku.</span>",
```
Comment (Anewbe) [inline code/game/objects/items/weapons/melee/energy.dm]: You need the \ at the end of this.

File: code/game/objects/items/weapons/melee/energy.dm [hunk 4]
```diff
@@ -58,9 +58,9 @@
 
 /obj/item/weapon/melee/energy/suicide_act(mob/user)
 	var/tempgender = "[user.gender == MALE ? "he's" : user.gender == FEMALE ? "she's" : "they are"]"
-	if (active)
-		viewers(user) << pick("<span class='danger'>\The [user] is slitting \his stomach open with \the [src]! It looks like [tempgender] trying to commit seppuku.</span>", \
-		                      "<span class='danger'>\The [user] is falling on \the [src]! It looks like [tempgender] trying to commit suicide.</span>")
+	if(active)
+		user.visible_message(pick("<span class='danger'>\The [user] is slitting \his stomach open with \the [src]! It looks like [tempgender] trying to commit seppuku.</span>",
+			"<span class='danger'>\The [user] is falling on \the [src]! It looks like [tempgender] trying to commit suicide.</span>"), "You start commiting suicide.")
```
Comment (Anewbe) [inline code/game/objects/items/weapons/melee/energy.dm]: That last one really shouldn't be there, the third input in a visible_message is a sound that is displayed to blind people, last I checked.
Also you need to end the span class.

File: code/game/objects/items/weapons/melee/energy.dm [hunk 5]
```diff
@@ -90,15 +90,15 @@
...
 
 /obj/item/weapon/melee/energy/axe/deactivate(mob/living/user)
 	..()
 	icon_state = initial(icon_state)
-	user << "<span class='notice'>\The [src] is de-energised. It's just a regular axe now.</span>"
+	to_chat(user, "<span class='notice'>\The [src] is de-energised. It's just a regular axe now.</span>")
 
 /obj/item/weapon/melee/energy/axe/suicide_act(mob/user)
-	viewers(user) << "<span class='warning'>\The [user] swings \the [src] towards \his head! It looks like \he's trying to commit suicide.</span>"
+	to_chat(viewers(user), "<span class='warning'>\The [user] swings \the [src] towards \his head! It looks like \he's trying to commit suicide.</span>")
```
Comment (Anewbe) [inline code/game/objects/items/weapons/melee/energy.dm]: This should be a visible_message.

File: code/game/objects/items/weapons/melee/energy.dm [hunk 6]
```diff
@@ -294,3 +294,73 @@
...
+ 	sharp = 1
+ 	edge = 1
+ 	force = 5
+ 	throwforce = 10
+ 	throw_speed = 7
+ 	throw_range = 11
+ 	reach = 2
+ 	w_class = ITEMSIZE_LARGE
+ 	active_force = 25
+ 	active_throwforce = 40
```
Comment (Anewbe) [inline code/game/objects/items/weapons/melee/energy.dm]: Why is this 40?

Comment (MisterLayne) [reply code/game/objects/items/weapons/melee/energy.dm]: I don't do the balance, I just put down the numbers. What would you suggest? Just copy-paste stats from the regular spear?

File: code/game/objects/items/weapons/melee/energy.dm (hunk 4 above)
Comment (MisterLayne) [reply code/game/objects/items/weapons/melee/energy.dm]: Span class is ended, on both quotes.

File: code/game/objects/items/weapons/melee/energy.dm (hunk 6 above)
Comment (Anewbe) [reply code/game/objects/items/weapons/melee/energy.dm]: 30 for active_throwforce, maybe?

File: code/game/objects/items/weapons/melee/energy.dm (hunk 4 above)
Comment (MisterLayne) [reply code/game/objects/items/weapons/melee/energy.dm]: There are only two inputs here. There's the 'pick' between the two suicide messages, and then there's the second, which is the message you see as the person.

**Comment by MisterLayne** [comment]
<iframe width="420" height="315"
src="https://www.youtube.com/embed/XGSy3_Czz8k">
</iframe>

**Pull request closed without merging**